src/service/crypto-verifier-impl.js

```diff
@@ -0,0 +1,198 @@
+/**
+ * Copyright 2016 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {dev} from '../log';
+import {getService} from '../service';
+import {
+  encodeByteArray,
+} from
+  '../../third_party/closure-library/encode-byte-array-trampoline-generated';
```

Comment:
The [manually managed dependencies](https://github.com/google/amphtml/blob/da40cb4339e0330751626c3e16a97a1905ef1f9b/third_party/closure-library/compile-encode-byte-array-trampoline.sh) in the trampolines files are a bit scary to me, especially when we can write a `encodeByteArray` ourselves. What are the size/speed tradeoffs we're going for with this?

```js
function encodeByteArray(byteArray) {
  var chars = Array(byteArray.length);
  for (var i = 0; i < byteArray.length; i++) {
    chars[i] = String.fromCharCode(byteArray[i]);
  }
  return btoa(chars.join(''));
}